code/modules/mob/living/simple_animal/hostile/zombie.dm

```diff
@@ -51,6 +51,14 @@
 			visible_message("<span class='danger'>[src] tears [L] to pieces!</span>")
 			src << "<span class='userdanger'>You feast on [L], restoring your health!</span>"
 			revive(full_heal = 1)
+			
+	if(istype(target, /obj/machinery/door/airlock))
+		src << "<span class='notice'>You start tearing apart the airlock...</span>"
+		playsound(src.loc, 'sound/hallucinations/growl3.ogg', 50, 1)
+		if(do_after(src, 250, target))
+			playsound(src.loc, 'sound/hallucinations/far_noise.ogg', 50, 1)
+			qdel(target)
```

Comment:
this, except ```new(src.loc)``` becomes ```new airlock.doortype(target.loc)```
1. Airlocks have specific assemblies, so they don't all look generic
2. src.loc would put it where the zombie is, that's weird